Update docker-compose.yml

ChristofferAAU opened the pull request:
Made it so that each environment has a separate name

# Description

Changed the names of the different environment names to match.

Feature/3env

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

ChristofferAAU committed:
Update docker-compose.yml

Made it so that each environment has a separate name

ChristofferAAU committed:
Updated the docker compose file for direct access

The container for the database is now accessible directly from the console on the specified port.

ChristofferAAU committed:
Updated to have a MYSQL root host

ChristofferAAU committed:
Updated so that test runs against latest and not dev

Brandhoej commented on .vs/ProjectSettings.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
+}
```
Burde .vs ikke være en del a .gitignore?

ChristofferAAU replied:
Der er ikke en .gitignore

Brandhoej replied:
Så synes jeg der skal laves en

Brandhoej replied:
Jeg kan ikke se, hvordan dette er blevet løst?

MagnusMat replied:
Nej det kan jeg heller ikke lige gennemskue.

Brandhoej commented on docker-compose.yml:
```diff
@@ -58,12 +58,12 @@ services:
             - backend
         environment:             # Sets local envionment for dotnet
-            ASPNETCORE_ENVIRONMENT: Production
+            ASPNETCORE_ENVIRONMENT: Staging
```
Hvis det er at miljøet er "Staging". Burde service navnet så ikke ændre fra "API_TEST" til "API_STAGE"?

Brandhoej commented on docker-compose.yml:
```diff
@@ -78,11 +78,16 @@ services:
+        expose:
+            - '3306'
         command: --default-authentication-plugin=mysql_native_password
         volumes:                 # Mounts the mysql files from the NFS to the container
             - /swarm-nfs/mysql/:/var/lib/mysql/
         environment:             # MySQL root password
-            MYSQL_ROOT_PASSWORD: Giraf123
+            - 'MYSQL_ROOT_PASSWORD=Giraf123'
+            - 'MYSQL_ROOT_HOST=%'
```
Når jeg ser dokumentation igennem ser jeg ikke denne måde at skrive environment variables med, og jeg vil mene at det vil være bedre, hvis vi holder måden til hvad der også er brugt ved de andre services.
```
- MYSQL_ROOT_PASSWORD=Giraf123
- MYSQL_ROOT_HOST=%
```
Jeg tror også det vil være en god ide at skrive hvorfor man bruger "%" til ROOT_HOST her, da det vil hjælpe de næste.

Dokumentation kan findes [her](https://docs.docker.com/compose/compose-file/compose-file-v3/#environment) og [her](https://docs.docker.com/compose/extends/) - for environment variables.

ChristofferAAU replied:
Hvorfor % bliver brugt vil blive lagt til wikien, og ROOT_HOST og ROOT_PASSWORD er blevet ændret til at passe med dokumentationen

Brandhoej replied:
Det er godt at % bliver skrevet om i wikien, men jeg vil mene at det er bedre at det er mere nært i forhold til koden.

ChristofferAAU committed:
The appsettings files now match the expected value

The appsettings file didn't match the names set, this has been changed with this commit. Develop has been changed to development and Testing has been changed to Staging.

ChristofferAAU committed:
MAde a few changes

ChristofferAAU committed:
Merge branch 'feature/3env' of https://github.com/aau-giraf/giraf-production-swarm into feature/3env

ChristofferAAU committed:
Updated the ROOT_HOST and ROOT_PASSWORD

ChristofferAAU committed:
Removed a password

Brandhoej commented on docker-compose.yml:
```diff
@@ -78,11 +78,16 @@ services:
         image: mysql:8.0.19
+        ports:
+            - '3306:3306'
+        expose:
+            - '3306'
```
Hvorfor bliver den samme port både publiseret ("ports") og exposed. Jeg vil mene det kun er nødvendigt med "ports".

HowToYEET replied:
rettet 🗡️

HowToYEET committed:
update yml (removed expose port in db)

HowToYEET committed:
deleted instead of gitignore  (not relevant)

JakobFaarGreg reviewed (approved):
LGTM

Sijmon reviewed (approved):
Godt arbejde champ, keep up the good work!
